### DISCUSSION TITLE
unmet peer dependencies when installing @trpc/server @trpc/client @trpc/react-query @trpc/next @tanstack/react-query zod

### DISCUSSION THREAD

**Discussion by saram7619** [trpc/trpc#5002] [Q&A]
I am using next.js 14 in a sass project, but I keep running into a peer dependency issue. I ran pnpm add @trpc/server @trpc/client @trpc/react-query @trpc/next @tanstack/react-query zod. In my vs code terminal I got the following warning.                               WARN  Issues with peer dependencies found
.
├─┬ @trpc/react-query 10.43.1
│ └── ✕ unmet peer @tanstack/react-query@^4.18.0: found 5.7.0
└─┬ @trpc/next 10.43.1
  └── ✕ unmet peer @tanstack/react-query@^4.18.0: found 5.7.0

Running pnpm install @tanstack/react-query@4.18.0 fixes this warning, then running 
 pnpm add @trpc/server @trpc/client @trpc/react-query @trpc/next @tanstack/react-query zod

Is there a way to install the packages without having the warning about unmet peer and installing an earlier version?

**Comment by medv** [comment]
@tanstack/react-query has been very recently updated to version 5.

@trpc/react-query requires it to be at version 4.18.0.

Until @trpc/react-query is updated to require the latest @tanstack/react-query, you must specifically install version 4 (`pnpm install @tanstack/react-query@^4.18.0`). The need for this will go away soon without you needing to do anything. You can inspect the package.json of @trpc/react-query and see its `peerDependencies` object (https://github.com/trpc/trpc/blob/main/packages/react-query/package.json#L65). When you see 5 make an appearance, it is safe to just install the latest react-query.
